ihmc-high-level-behaviors/src/main/java/us/ihmc/behaviors/activeMapping/StancePoseCalculator.java

```diff
@@ -3,57 +3,43 @@
 import us.ihmc.euclid.referenceFrame.FramePose3D;
-import us.ihmc.euclid.tools.EuclidCoreTools;
 import us.ihmc.euclid.transform.RigidBodyTransform;
-import us.ihmc.euclid.tuple3D.interfaces.UnitVector3DBasics;
 import us.ihmc.euclid.tuple3D.interfaces.UnitVector3DReadOnly;
-import us.ihmc.euclid.tuple3D.interfaces.Vector3DReadOnly;
 import us.ihmc.footstepPlanning.graphSearch.FootstepPlannerEnvironmentHandler;
 import us.ihmc.footstepPlanning.polygonSnapping.HeightMapPolygonSnapper;
+import us.ihmc.footstepPlanning.polygonSnapping.PolygonSnapperTools;
 import us.ihmc.perception.heightMap.TerrainMapData;
 import us.ihmc.robotics.robotSide.RobotSide;
 import us.ihmc.robotics.robotSide.SideDependentList;
-import us.ihmc.sensorProcessing.heightMap.HeightMapData;
 
 import java.util.ArrayList;
 
 public class StancePoseCalculator
 {
-   public static final float CONTACT_COST_WEIGHT = 10.0f;
-   public static final float MAX_CONTACT_VALUE = 255.0f;
-   public static final float NOMINAL_STANCE_DISTANCE = 0.5f;
+   private final StancePoseParameters stancePoseParameters = new StancePoseParameters();
 
-   private int windowSize = 5;
-   private double resolution = 0.08f;
-   private double maxWidth = 0.5f;
-   private double maxLength = 0.5f;
-   private double maxYaw = 0.1f;
+   private final HeightMapPolygonSnapper heightMapPolygonSnapper;
 
-   private HeightMapPolygonSnapper heightMapPolygonSnapper;
+   private final ArrayList<FramePose3D> leftPoses = new ArrayList<>();
+   private final ArrayList<FramePose3D> rightPoses = new ArrayList<>();
 
-   private ArrayList<FramePose3D> leftPoses = new ArrayList<>();
-   private ArrayList<FramePose3D> rightPoses = new ArrayList<>();
-
-   private SideDependentList<Pose3D> bestPose3Ds = new SideDependentList<>(new Pose3D(), new Pose3D());
-   private SideDependentList<FramePose3D> bestFramePoses = new SideDependentList<>(new FramePose3D(), new FramePose3D());
+   private final SideDependentList<Pose3D> bestPose3Ds = new SideDependentList<>(new Pose3D(), new Pose3D());
+   private final SideDependentList<FramePose3D> bestFramePoses = new SideDependentList<>(new FramePose3D(), new FramePose3D());
 
    private final SideDependentList<ConvexPolygon2D> footPolygons;
 
-   public StancePoseCalculator(double maxWidth, double maxLength, double maxYaw, SideDependentList<ConvexPolygon2D> footPolygons)
+   public StancePoseCalculator(SideDependentList<ConvexPolygon2D> footPolygons)
    {
-      this.maxWidth = maxWidth;
-      this.maxLength = maxLength;
-      this.maxYaw = maxYaw;
       this.footPolygons = footPolygons;
       this.heightMapPolygonSnapper = new HeightMapPolygonSnapper();
    }
 
-   public SideDependentList<FramePose3D> getStancePoses(FramePose3D goalPose, TerrainMapData terrainMap, FootstepPlannerEnvironmentHandler environmentHandler)
+   public SideDependentList<FramePose3D> getStancePoses(FramePose3D pickPoint, TerrainMapData terrainMap, FootstepPlannerEnvironmentHandler environmentHandler)
```

Comment:
I wouldn't call this a pickpoint, I'd call it the `midstancePose` or something like that. PickPoint implies this only comes form selection.

Reply:
You can also make this a `FramePose3DReadOnly`, which I think would be a lot safer. Additionally, it's highly likely the returns could be `SideDependentList<? extends FramePose3DReadOnly>`

Reply:
Yep all good points

Reply:
Couldn't get the method to return SideDependentList<? extends FramePose3DReadOnly>